src/thx/schema/macro/BoundSchemaType.hx

```diff
@@ -0,0 +1,155 @@
+package thx.schema.macro;
+
+import haxe.macro.Context;
+import haxe.macro.Expr;
+import haxe.macro.ExprTools;
+import haxe.macro.Type;
+import haxe.macro.TypeTools;
+import thx.schema.macro.Error.*;
+import thx.schema.macro.Utils.*;
+import haxe.ds.Option;
+using thx.Options;
+
+// this is the type for things inside object literals used like this: Generic.schema({ name: Option<String> })
+// and also it is used when looking up schemas during generation
+class BoundSchemaType {
```

Comment:
Yeah potentially. The `UnboundSchemaType` is probably not even necessary and I have to try to remove it.